volatility3/framework/symbols/windows/extensions/__init__.py

```diff
@@ -775,39 +776,131 @@ def get_peb(self) -> interfaces.objects.ObjectInterface:
+        # We use the WindowsKernelIntermedSymbols class to make
+        # sure we get all the object helpers. For example, traversing
+        # linked-lists.
+        self._32bit_table_name = windows.WindowsKernelIntermedSymbols.create(
+            self._context, config_path, "windows", "wow64"
+        )
+
+        # windows 10
+        if self._context.symbol_space.has_type(
+            sym_table + constants.BANG + "_EWOW64PROCESS"
+        ):
+            offset = proc.Peb
+
+        # vista sp0-sp1 and 2003 sp1-sp2
+        elif self._context.symbol_space.has_type(
+            sym_table + constants.BANG + "_WOW64_PROCESS"
+        ):
+            offset = proc.Wow64
+
+        else:
+            offset = proc
+
+        peb32 = self._context.object(
+            f"{self._32bit_table_name}{constants.BANG}_PEB32",
+            layer_name=proc_layer_name,
+            offset=offset,
+        )
+        return peb32
+
     def load_order_modules(self) -> Iterable[interfaces.objects.ObjectInterface]:
         """Generator for DLLs in the order that they were loaded."""
-
         try:
-            peb = self.get_peb()
-            yield from peb.Ldr.InLoadOrderModuleList.to_list(
-                f"{self.get_symbol_table_name()}{constants.BANG}_LDR_DATA_TABLE_ENTRY",
-                "InLoadOrderLinks",
-            )
+            pebs = [
+                self.get_peb(),
+                self.get_peb32(),
+            ]
+            for peb in pebs:
+                if peb:
+                    sym_table = self.get_symbol_table_name()
+                    if peb.Ldr.vol.type_name.endswith("unsigned long"):
+                        ldr_data = self._context.symbol_space.get_type(
+                            self._32bit_table_name + constants.BANG + "_PEB_LDR_DATA"
+                        )
+                        peb.Ldr = peb.Ldr.cast("pointer", subtype=ldr_data)
+                        sym_table = self._32bit_table_name
+                    yield from peb.Ldr.InLoadOrderModuleList.to_list(
+                        f"{sym_table}{constants.BANG}" + "_LDR_DATA_TABLE_ENTRY",
+                        "InLoadOrderLinks",
+                    )
         except exceptions.InvalidAddressException:
             return None
 
     def init_order_modules(self) -> Iterable[interfaces.objects.ObjectInterface]:
         """Generator for DLLs in the order that they were initialized"""
 
         try:
-            peb = self.get_peb()
-            yield from peb.Ldr.InInitializationOrderModuleList.to_list(
-                f"{self.get_symbol_table_name()}{constants.BANG}_LDR_DATA_TABLE_ENTRY",
-                "InInitializationOrderLinks",
-            )
+            pebs = [
+                self.get_peb(),
+                self.get_peb32(),
+            ]
+            for peb in pebs:
+                if peb:
+                    sym_table = self.get_symbol_table_name()
+                    if peb.Ldr.vol.type_name.endswith("unsigned long"):
```

Comment:
Hard coding specific types in may be troublesome in the future.  There was an instance in the linux world where the pointers when from `unsigned long` to `long unsigned int` because of a change in Clang.  This also just verifies the end, rather than everything after the `constants.BANG` so you might catch unwanted/unexpected types.  Probably better to do something like `constants.BANG.split(...vol.type_name)[-1] == 'unsigned long'` although that would need testing to make sure it works properly in all instances.

Reply:
Would there be a big problem in just always casting the value, regardless of whether it's already the right type, or just verify it's a point to the right subtype and otherwise cast it?